amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -264,43 +264,82 @@ def write_micro_benchmark_simpleion(iterations, warmups, c_extension, obj, file,
-def write_micro_benchmark_event(iterations, warmups, c_extension, obj, binary, file=None):
-    pass
+def write_micro_benchmark_event(iterations, warmups, c_extension, obj, file, binary, memory_profiling):
+    return 0, 0
 
 
 # Framework for benchmarking write methods, this functions includes
 # 1. profile memory usage,
 # 2. benchmark performance,
 # 3. generate report
-def write_micro_benchmark_and_profiling(write_micro_benchmark_function, iterations, warmups, obj, c_extension, binary, file):
+def write_micro_benchmark_and_profiling(table, write_micro_benchmark_function, iterations, warmups, obj, c_extension,
+                                        binary, file, each_option):
     if not obj:
         raise Exception("Invalid obj: object can not be none.")
     if not write_micro_benchmark_function:
         raise Exception("Invalid micro benchmark function: micro benchmark function can not be none.")
     # Memory Profiling
     if not pypy:
-        write_micro_benchmark_function(iterations=1, warmups=0, obj=obj, c_extension=c_extension, file=file, binary=binary,
+        write_micro_benchmark_function(iterations=1, warmups=0, obj=obj, c_extension=c_extension, file=file,
+                                       binary=binary,
                                        memory_profiling=True)
 
     # Performance Benchmark
     file_size, result_with_gc = \
-        write_micro_benchmark_function(iterations=iterations, warmups=warmups, obj=obj, c_extension=c_extension, file=file,
+        write_micro_benchmark_function(iterations=iterations, warmups=warmups, obj=obj, c_extension=c_extension,
+                                       file=file,
                                        binary=binary, memory_profiling=False)
 
     # generate report
-    write_generate_report(file_size, result_with_gc, write_memory_usage_peak)
+    write_generate_report(table, file_size, each_option, result_with_gc, write_memory_usage_peak)
 
-    return file_size, result_with_gc, write_memory_usage_peak
+    return file_size, each_option, result_with_gc, write_memory_usage_peak
 
 
 # Generates and prints benchmark report
-def write_generate_report(file_size, total_time, memory_usage_peak):
-    table = [['file_size (MB)', 'total_time (s)', 'memory_usage_peak (MB)'],
-             [format_decimal(file_size),
-              format_decimal(total_time),
-              format_decimal(memory_usage_peak)]]
-    print('\n')
-    print(tabulate(table, tablefmt='fancy_grid'))
+def write_generate_report(table, file_size, each_option, total_time, memory_usage_peak):
+    insert_into_report_table(table,
+                             [format_decimal(file_size),
+                              each_option,
+                              format_decimal(total_time),
+                              format_decimal(memory_usage_peak)])
+
+
+# Insert a benchmark result row into the benchmark report (table)
+def insert_into_report_table(table, row):
```

Comment:
We now create a report table according to the execution command first, and then insert benchmark result row into the table after each execution.